### DISCUSSION TITLE
[Feature] Secondary server url

### DISCUSSION THREAD

**Discussion by tobiasdiez** [immich-app/immich#14186] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I'm currently hosting immich on a private home server that is not exposed to the internet. To access it remotely, I'm using tailscale on my devices. This setup works nicely, also for image backup via immich.

However, I have to specify the tailscale url for the server in order for immich to communicate with it. Thus, even if I'm on the same local netwok (= home), I have to activate tailscale for the backup to work. It would be nice if one could instead specify a secondary or local url, that is used as fall-back in case the first url of the server cannot be accessed.

Something like this is possible with home assistant which allows to specify an "internal and external" url (see eg https://companion.home-assistant.io/docs/troubleshooting/networking?_highlight=internal&_highlight=url#setting-up-the-app)

Duplicate of https://github.com/immich-app/immich/issues/308, which is closed although the proposed solution in this thread does not work (you cannot change the port via a local dns).

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by Mraedis** [comment]
Wouldn't it be easier if you did local split DNS (rewrites) to that local calls still use the external URL but with an internal IP?

**Reply by tobiasdiez** [reply]
Yes, that might work if both services use the same port. Specifying a second url is a) more flexible and b) easier to setup in most scenarios.

**Comment by bo0tzz** [comment]
You're already aware that this is a duplicate, so why open a new thread?

**Reply by tobiasdiez** [reply]
Because it was closed with the reasoning that one can use a local dns to solve it. But this doesn't always work, as one cannot change the port via dns rewrites.

**Reply by Mraedis** [reply]
It would make more sense to use :80 or :443 behind a proxy and route based on subdomain. (Like how everyone, everywhere does it 🙂. )
